# eDNA metabarcoding, pipeline development & high performance computing

***MSc course in "Applied Bioinformatics & Data Analysis"***


<img src="https://camo.githubusercontent.com/762c1129f266494bbbb3faff3d673040cf7b1f19d45c6e13f49b08de12f5116a/68747470733a2f2f692e70617374652e706963732f38373031383966616466363638613935386338616163383366333865373939632e706e67" width= 220px; align="left"  />


<br/><br/><br/>

PEMA ([*Zafeiropoulos et al.* 2020](https://academic.oup.com/gigascience/article/9/3/giaa022/5803335))
supports the complete analysis of 4 marker genes: 16S/18S rRNA, ITS and COI.
In its latest version a new module was added to enable the use of custom reference databases. 

You may find more about PEMA on its [documentation site](http://pema.hcmr.gr/)
but also on its [GitHub repo](https://github.com/hariszaf/pema). 


You may have a look on PEMAv2 on its recent presentation at [DNAQUAnet2021](https://www.youtube.com/watch?v=kht_LKMmB6w).


<br/><br/>

## MSc class course in "Applied Bioinformatics & Data Analysis" 

The tasks need to be performed in the framework of the class can be found in the [msc_democr_uni.md](https://github.com/hariszaf/pema-tutorials/blob/ms-democr-class/msc_democr_uni.md).


## PEMA as an open source project

If you're interested in contributing at the code level, you may have a look [here](https://github.com/hariszaf/pema/blob/master/CONTRIBUTING.md). 
Keeping PEMA up-to-date, adding new modules and alternatives is an iterative process and you are the most welcome to join this effort!

To contact PEMA team you can find us through the [PEMA's Gitter community](https://gitter.im/pema-helpdesk/community) or directly at: pema@hcmr.gr. 

